packages/x-date-pickers/src/tests/describeValue/testControlledUnControlled.tsx

```diff
@@ -51,14 +64,43 @@ export const testControlledUnControlled: DescribeValueTestSuite<any, any> = (
       const newValue = setNewValue(values[0]);
 
       expect(onChange.callCount).to.equal(componentFamily === 'clock' ? 2 : 1);
-      // TODO: Support range
-      expect(onChange.lastCall.args[0]).toEqualDateTime(newValue as any);
+      if (Array.isArray(newValue)) {
+        newValue.forEach((value, index) => {
+          expect(onChange.lastCall.args[0][index]).toEqualDateTime(value);
+        });
+      } else {
+        expect(onChange.lastCall.args[0]).toEqualDateTime(newValue as any);
+      }
     });
 
     it('should react to `props.value` update', () => {
       const { setProps } = render(<ElementToTest value={values[0]} />);
       setProps({ value: values[1] });
       assertRenderedValue(values[1]);
     });
+
+    ['readOnly', 'disabled'].forEach((prop) => {
+      it(`should apply ${prop}="true" prop`, () => {
+        if (!['field', 'new-picker'].includes(componentFamily)) {
+          return;
+        }
+        const handleChange = spy();
+        render(<ElementToTest value={values[0]} onChange={handleChange} {...{ [prop]: true }} />);
+
+        // assert that editing in "readOnly" mode is not allowed
```

Comment:
I assume the reason to add such a test is that event with the `readOnly` props it's feasible to edit dates with arrowUp/arrowDown

If that's the reason it would be nice to explain it in the comment such that a new dev in 1 year can understand why this test is here

Reply:
I was debating adding it in, but the currently existing issue pushed me towards adding it in. 🤔 

This case would probably be caught by the missing `readonly` attribute and might be redundant... 🤔 
Would you agree that removing it is for the best?

Reply:
I've [extracted the test](https://github.com/mui/mui-x/pull/7135/files/#diff-5cdf0facdbadac7fca4b02cd2f21e4e274e2911f215483ca1b9fe0906faf3bebR98) for mobile/readonly keyboard editing. 😉